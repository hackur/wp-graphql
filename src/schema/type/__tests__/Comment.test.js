import CommentType from 'type/Comment';

// eslint-disable-next-line no-underscore-dangle
const fields = CommentType._typeConfig.fields();

describe('Test Comment type', () => {
  test('Test name', () => {
    expect(CommentType.name).toMatchSnapshot();
  });

  test('Test fields', () => {
    expect(Object.keys(CommentType.getFields())).toMatchSnapshot();
  });

  test('Test resolve post', () => {
    const post = fields.post.resolve({ post: 13 });
    expect(post).toMatchSnapshot();
  });

  test('Test resolve parent', () => {
    const parent = fields.parent.resolve({ parent: 69 });
    expect(parent).toMatchSnapshot();
  });

  test('Test resolve parent null', () => {
    expect(fields.parent.resolve({ parent: 0 })).toBeNull();
  });

  test('Test resolve avatars', () => {
    const avatars = fields.author_avatar_urls.resolve({
      author_avatar_urls: {
        thumbnail: 'https://scott.com/avi.jpg',
      },
    });
    expect(avatars).toMatchSnapshot();
  });
});
