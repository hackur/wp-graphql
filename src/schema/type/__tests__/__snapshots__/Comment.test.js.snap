// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Comment type Test fields 1`] = `
Array [
  "id",
  "author",
  "date",
  "date_gmt",
  "content",
  "link",
  "post",
  "parent",
  "author_name",
  "author_url",
  "author_hash",
  "status",
  "type",
  "author_avatar_urls",
  "meta",
]
`;

exports[`Test Comment type Test name 1`] = `"Comment"`;

exports[`Test Comment type Test resolve avatars 1`] = `
Array [
  Object {
    "size": "thumbnail",
    "url": "https://scott.com/avi.jpg",
  },
]
`;

exports[`Test Comment type Test resolve parent 1`] = `"Q29tbWVudDo2OQ=="`;

exports[`Test Comment type Test resolve post 1`] = `"UG9zdDoxMw=="`;
